# Protein-coding genes {#lesson7}

## Learning Objectives

* Identify the number of protein-coding gene
* Explain how protein-coding genes are distributed along chromosomes.
* Show how to determine the function of protein-coding genes.
* Annotate genomes using both ab initio and evidence-based approaches
* Apply ab initio and evidence-based approaches to annotate genomes.

## Assignment

For this assignment and following assignments I will be looking at the genome of *[Chromera velia](https://en.wikipedia.org/wiki/Chromera_velia)*, a recently described unicellular eukaryote closely related phylogenetically to the Apicomplexa phylum of protozoan parasites. In a paper by [Woo et al. 2015](http://elifesciences.org/content/4/e06974v1), they sequence and assemble the nuclear genomes of two, so-called, Chromerids, *Chromera velia* and *Vitrella brassicaformis*. Key questions revolve around the evolutionary emergence of the Apicomplexa considering the Chromerids represent the closest non-Apicomplexan relative. Here will only focus on *Chromera*.

I will breifly describe its genome based on the paper mentioned above and attempt to use [Augustus](http://nar.oxfordjournals.org/content/34/suppl_2/W435) to predict gene sequences.

**A brief look at the Chromerid genome**

It's important to note that the assembly I will be working is a draft assembly. It is made of 5953 scaffolds and totals 193.6 million base-pairs (Mb). The predicted number of protein-coding genes is 26,112 and the genome is estimated to be composed of ~30 Mb in transposable elements (TEs). This is an interesting finding considering that Plasmodium, for example, is known to have an exceptional dearth in transpoable elements. Thus, either the transposable elements that were present in the proto-Apicomplexan ancestor have been lost or repurposed. If we look at Supplementary table 2 in Woo et al., they note that 22.82% of the TEs fall under "Unclassified repeats". This suggests the potential for novel TEs awaiting discovery within protist and potentially Apicomplexan genomes.

Interestingly, in *Vitrella*, while the proporition of putative TEs is similar as in  *Chromera* (27.24%), the genome is much smaller at 72.7 Mb. In addition, the *Vitrella* genome was assembled into much fewer scaffolds at 1064. Lastly, the number of protein-coding genes was not drastically different at 22,817 suggesting that there is far less intergenic space in *Vitrella*, similar to that in parasitic genomes of Apiceomplxan parasites.

**Choosing a gene of interest**

The genomic loci of interest in this study is going to be ~30kb on scaffold 27 of the *Chromera* assembly. This regions represents 10kb flanks surroundg the gene identified as Cvel_872. This gene was assigned the function of a striated fiber assemblin (SFA) protein, typically associated with basal body rootlets of the flagella. While all known components of the flagella were present in proto-apicomplexan ancestors (based on phyolgenetic analysis), most of the components have been lost as apicomplexan lineages began to differentiate. Thus, it's been hypothesized that some flagellar-proteins have been repurposed for new functions in apicomplexans ([Francia et al. 2012](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001444)).

In addition, Woo et al. preformed RNA-seq and compared the expression profiles of all genes in *Chromera* to all genes in *Plasmodium falciparum*. A statistically significant overlap in highly co-expressed orthogroups (orthologous gene clusters identified through sequence comparison) between *falciparum* invasion genes and the *Chromera* gene encoding SFA. Thus, it suggests an interesting hypothesis that elements of the flagellar apparatus have been repurposed as a module of the apical complex, an important organelle for invading host cells found only in Apicomplexan parasites.

Here, I chose to use Augustus to predict the genes within 10kb nucleotide flanking regions up and downstream of the gene sequence encoding SFA itself, including the sequence of the gene encoding SFA, itself.

**Evaluating Augustus**

In Woo et al., they combined several approaches to predict genes in *Chromera*. They specifically used Augustus, a gene prediction software tool based on a Generalized Hidden Markov Model. In order to train the HMM they used Cufflinks derived transcript assemblies obtained from the RNA-seq experiments. If you don't have transcript models to train the HMM, Augustus comes with dozens fo pretrained HMMs based on gene models in other organisms.

Here I use *Toxoplasm gondii* as my gene model prediction organism. The snapshot in IGV represents my results.

```{r results, echo=F, fig.cap="Augustus Gene Model Predictions. Gene models predicted in Woo et al. are in black; Augustus predictions without hints are in blue; Augustus preidctions with hints are in green.", fig.align="center"}
knitr::include_graphics("img/lesson7/results.png")
```

The red box represents the gene encoding SFA. What can see here is that Augustus is very good at predicting gene models *ab initio*. There is virtually no difference between the output when you compare the output with hints and without. There is, however, one general trend that plagues the accuracy of prediction, and that is premature stop codons, shown in Figure 2 below.

```{r errors, echo=F, fig.cap="Example of using the wrong reading frame to predict a stop codon.", fig.align="center"}
knitr::include_graphics("img/lesson7/errors_in_prediction.png")
```

In Figure 2 we can see that Augustus chose to predict a CDS that doesn't exist (the start of the thick bar) and terminate the CDS (start of the thin bar) prematurely due to an incorrect reading frame prediction. With alternatively spliced transcripts like these this is likely a common problem. However, overall it does a great job! Looking at Figure 1, we can confidentally say that have a generally idea of where genes are, how many exons they contain, and roughly where splice site junctions are located.

We can use a program called [gffcompare](https://ccb.jhu.edu/software/stringtie/gff.shtml#gffcompare) to quantify the similarity in our predictions. Accordingly, these predictions are about 70.5% accurate. The reason this number is lower than expected when looking at the figures is due to the partial predictions of genes at the edges. I did not choose the flanking regions with the knowledge of where genes were upsteam and downstream of SFA. Thus, the predictions can only go so far as to predict partial genes at the edges, which it does relatively well.

**Conclusions**

Augustus is a very accurate and useful tool to predict protein-coding genes in eukaryotic genomes, given the correct HMM training model.
