# Comparative genomices {#lesson9}

## Learning Objectives

* Distinguish neutral evolution, negative selection, and positive selection
* Interpret alignments of genomic DNA sequences as showing evidence of evolutionary constraint or adaptation, distinguishing both from evidence consistent with neutral evolution.
* Show how you can use genome comparisons between species to estimate the amount of functional sequence – and to identify it
* Understand the basics of several tests for selection or scores that estimate likelihood of being under selection
    * Ka/Ks test
    * phastCons
    * Extended haplotype homozygosity
* Understand the connection between gene duplication and adaptive evolution

## Notes

> "Functionally less important molecules or parts of a molecule evolve faster than more important ones."
> — Kimura and Ohta (1974) PNAS USA 71: 2848-2852

**Fundamental process in comparative genomics**

1. Get genome sequences from species or individuals separated by a distance appropriate to the question you are addressing
2. Align those sequences
3. Find informative similarities
  - BLAST search or sequences databases
4. Compare the alignments to a neutral model or appropriate group
  - Amount of similarity
  - Likelihood of being under selection
  - Patterns in alignments

## Assignment

### Background

In Woo et al. [@chromerids] they make note of a statistically significant relationship between a gene encoding striated fiber assemblin (SFA), Cvel_872, and several host-cell invasion protein encoding genes in *P. falciparum*. Their analysis showed correlated expression patterns between various orthologues gene groups, or orthogroups, looking at various apicomplexan species and the single-celled red algae, *Chromera velia*. The most significant overlap between orthogroups occured between *Chromera velia* and *P. falciparum*. SFA is a key protein for organizing the basal bodies of the flagellar apparatus in algae and the apical complexes in apicomplexans. Thus, they suggest that, in *Chromera velia*, some of the genes implicated in the invasion process of modern apicomplexans were functionally associated with those implicated in flagellar motility.

### Design

I decided to analyze the evolutionary signatures present in the homologue of Cvel_872 in various *Plasmodium* species. 

### Methods

The gene IDs for the those genes are as follows:

* PF3D7_0703600 (human)
* PBANKA_0801300 (mouse)
* PCHAS_0801600 (mouse)
* PCYB_011100 (macaque)
* PKNH_0101800 (macaque & human)
* PRCDC_0701800 (chimpanzee)
* PVX_087720 (human)
* PY17X_0804000 (mouse)

Vertebrate host species known to be infected by the above parasites with which each gene is associated with are in parentheses.

The sequences were extracted from [PlasmoDB](http://plasmodb.org/plasmo/) on July 11th, 2016 from the latest version (28). Following, data acquisition, rate of substitutions at synonymous and nonsynonymous sites were analyzed using the Ka/Ks Calculation tool found at [http://services.cbu.uib.no/tools/kaks](http://services.cbu.uib.no/tools/kaks).

### Results

#### Pairwise alignments

Using the EMBOSS Needle alignment program, I calculated the following protein sequence alignments for the gene in *Chromera velia* and *Plasmodium falciparum*.

```
#=======================================
#
# Aligned_sequences: 2
# 1: Pfalciparum
# 2: Cvelia
# Matrix: EBLOSUM62
# Gap_penalty: 10.0
# Extend_penalty: 0.5
#
# Length: 293
# Identity:      68/293 (23.2%)
# Similarity:   145/293 (49.5%)
# Gaps:          37/293 (12.6%)
# Score: 327.0
# 
#
#=======================================

Pfalciparum        1 ------------------------------MEELNKSGIYSKQFSLSQNK     20
                                                   :..|.:.|    ..:|:...
Cvelia             1 MSAPPSHSVSGPSGGLGVGGGMPMPSSSSSVSALAQQG----GGTLASPH     46

Pfalciparum       21 HNSERLQRLEKRLSGLHFSIELQKNEKIDKLNEKINSLEEKLIEMHENSN     70
                     |  :|:.:|.:|||||...:|::|:.:.:.|:.::.:::|:|....::|:
Cvelia            47 H--DRIAKLSQRLSGLQVGLEVEKSSRFEALHGRLRAIDERLSTTQDSSS     94

Pfalciparum       71 KTFEKLNEKLNDIRNDVTNYKNELEEFKNDHKKKLQLLEEKAEDFINKEK    120
                     |.|..|.|:|...:.::...:...|:...|.::.:.:|:::....:..|:
Cvelia            95 KRFSTLKEQLTAFQRELEEERGAREKLGEDRQRDVAVLDQRLSHTLESEQ    144

Pfalciparum      121 EDWSRLKIKLVKDFQHKAALLKEEMVEEYGLIEEKEDSLRKYYDCEINNI    170
                     :.....:.:|::.|:.|..:||:|:::|..|..|.|.:||:|.:.:|..:
Cvelia           145 QARRDAESRLLRAFEEKTKVLKDEVLKEGRLRVENESALRRYIEADIPKL    194

Pfalciparum      171 KSIIQNEISERIKTEKIILSDVDDKINEIMKIIRNEKTTRETYSENLVSL    220
                     ..:::.|.:.|...|..|:....::||.:.::::.||..||...|.|:.:
Cvelia           195 YEMLKEECTSREAMESRIVKKATEEINRLTELVQQEKKAREETEEALLRM    244

Pfalciparum      221 IEQYFSRIKKEIDMERLEREDTEETLVHLMEEALDKIGIPLA-    262
                     :|....::|.||..||.:||.|||||:.|:|:...|:..... 
Cvelia           245 LEDIVQKMKSEISQERSDREATEETLLRLLEDTCSKLNAVAQL    287
```

I then aligned the protein sequences for the same gene in *Plasmodium vivax* to the same sequence as above in *Plasmodium falciparum*.

```
#=======================================
#
# Aligned_sequences: 2
# 1: Pfalciparum
# 2: Pvivax
# Matrix: EBLOSUM62
# Gap_penalty: 10.0
# Extend_penalty: 0.5
#
# Length: 262
# Identity:     218/262 (83.2%)
# Similarity:   248/262 (94.7%)
# Gaps:           0/262 ( 0.0%)
# Score: 1133.0
# 
#
#=======================================

Pfalciparum        1 MEELNKSGIYSKQFSLSQNKHNSERLQRLEKRLSGLHFSIELQKNEKIDK     50
                     ||:|.||||||||||||||||:||||||||||||||||||||||||||||
Pvivax             1 MEDLAKSGIYSKQFSLSQNKHSSERLQRLEKRLSGLHFSIELQKNEKIDK     50

Pfalciparum       51 LNEKINSLEEKLIEMHENSNKTFEKLNEKLNDIRNDVTNYKNELEEFKND    100
                     ||||||.|||||:||:||||:|||||||||||||.||.|||.||||||||
Pvivax            51 LNEKINLLEEKLVEMNENSNRTFEKLNEKLNDIRVDVNNYKTELEEFKND    100

Pfalciparum      101 HKKKLQLLEEKAEDFINKEKEDWSRLKIKLVKDFQHKAALLKEEMVEEYG    150
                     |:|||::|::|||:||:||||:|::|||||:||||:|.:||||||.||||
Pvivax           101 HRKKLEILQKKAEEFIDKEKEEWNKLKIKLIKDFQNKTSLLKEEMTEEYG    150

Pfalciparum      151 LIEEKEDSLRKYYDCEINNIKSIIQNEISERIKTEKIILSDVDDKINEIM    200
                     |::|||:|||::||.||.|:||::||||:||||||||||.|||||:|||:
Pvivax           151 LLQEKEESLRRFYDNEIANLKSVVQNEINERIKTEKIILRDVDDKVNEII    200

Pfalciparum      201 KIIRNEKTTRETYSENLVSLIEQYFSRIKKEIDMERLEREDTEETLVHLM    250
                     |||:|||.|||.|||||||||||||||||||||.||||||||||||||||
Pvivax           201 KIIKNEKITRENYSENLVSLIEQYFSRIKKEIDNERLEREDTEETLVHLM    250

Pfalciparum      251 EEALDKIGIPLA    262
                     ||||||||||.|
Pvivax           251 EEALDKIGIPFA    262
```

#### Multiple sequence alignment

We can use the Bioconductor package, [msa](https://bioconductor.org/packages/release/bioc/html/msa.html) to visualize it for us here. We can see that we have very few loci that are variable, even at the DNA level, for this gene.

```{r msa-sfa,fig.align="center",fig.caption="Multiple sequence alignment of SFA homologs in selected Plasmodium species",results=""}
sshhh("msa")
seqs <- readDNAStringSet("_data/lesson9/OG5_136181_CDS.fasta")
aln <- msa(seqs)
print(aln, showNames=TRUE, show="complete")
```

#### Phylogenetic tree

The phylogenetic tree is consistent with the phylogeny estimated in Woo et al. and with what's known about *Plasmodium* evolution [@Liu_2010@Liu_2014].

```{r kaks-tree, echo=F, fig.cap="Annotated phylogenetic tree", fig.align="center"}
knitr::include_graphics("img/lesson9/kaks_tree.png")
```

#### Ka / Ks values

The first thing to consider when evaluating the Ka / Ks ratios are whether the alignment was made between genes that are 1:1 homologs of one another. While there are multiple SFA genes within some *Plasmodium* spcecies, such as *falcparum*, they underwent an expansion prior to the the evolution of the proto-*Plasmodium* ancestor. Thus, these genes represent 1:1 homologs when comparing only *Plasmodium* species.

The second thing we need to consider is the evolutionary distance between the species being compared. While the exact evolutionary emergence of different *Plasmodium* species is still up for debate, we know that the Prioplasmida and *Plasmodium* families diverged ~ 300 mya [@SILVA_2010]. We also know that the *Plasmodium* lineages that infect mammals emerged ~34 mya [@Pacheco_2013]. However, the times of evolutionary divergence between the species used in this analysis varies. *P. falciparum* for example, is most recently estimated to have emerged only 10,000 years ago whereas *P. vivax* is estimated to be over 1 mya.

```{r kaks-table, echo=F, fig.cap="Calculated Ka / Ks ratios for an SFA-derived gene in Plasmodium parasites", fig.align="center"}
knitr::include_graphics("img/lesson9/kaks_table.png")
```

### Discussion

Based on the results above, I conlcude that the homologs of Cvel_872, or SFA, are being acted upon by purifying selection. Although the multiple sequence alignment shown above is expected when comparing homologs within the same genus, *Plasmodium* species are only ~85% alignable. Thus, sequences that are substantially alignable between even species, often indicates functional importance. The last common apixomplexan ancestor is estimated to have existed ~350-824 mya, so the alignment similiarity between *P. falciparum* and *C. velia* supports this claim as well.

